llvm/lib/Transforms/Vectorize/VPlan.cpp

```diff
@@ -514,6 +546,11 @@ void VPBasicBlock::execute(VPTransformState *State) {
 
   // 2. Fill the IR basic block with IR instructions.
   executeRecipes(State, NewBB);
+
+  // If this block is a latch, update CurrentParentLoop.
```

Comment:
```suggestion
  // If this block is a latch, update CurrentParentLoop. The second successor of a latch is a header, with the other successor being an exit/middle.
```
Wrapping in `VPBlockUtils::isLatch()` may be clearer?

Reply:
done thanks